### PR TITLE
HFP-2992 Adjust to meet design created after pull request

### PR DESCRIPTION
**Pull request by otacke**
- Move requiresFinishing setting in semantics.json
- Adjust path to requiresFinishing in view
- Add missing fields in translations

### CODE REVIEW

**Comment by simondate** [comment]
Can you remove this please? https://github.com/otacke/h5p-branching-scenario/blob/HFP-2992-adjust-to-design/semantics.json#L228-L245

**Review by simondate — COMMENTED** [review]
Why aren't the changes consistent between all the different language files?

**Review by simondate — CHANGES_REQUESTED** [review]
I constantly get a breaking error when trying to press Proceed after completing a IV in the preview mode. Something to do with firing an xAPI statement that it can't handle. The error doesn't occur when viewing it normally.

Screenshot 2021-01-25 at 11 58 23

File: language/.en.json
Comment (otacke) [reply language/.en.json:61]: Just what the spec said ...

**Comment by otacke** [comment]
> Can you remove this please? https://github.com/otacke/h5p-branching-scenario/blob/HFP-2992-adjust-to-design/semantics.json#L228-L245

@simondate Before I comment, please let me rest you assured that I know you just joined the band :-)

I can remove that, of course, but the whole idea of the pull-request (again: it was pre-qualified with Svein-Tore) was to provide a general solution that would work for any content type, not just IV and CP that just need some extra care because of the way they determine when a content is completed. It does. You can put in Fill-in-the-Blanks or whatever has a completed statement and you can prevent progression for those, too. The designs in HFP-2992 and HFP-2993 were created over half a year later than the pull request was filed (https://h5ptechnology.atlassian.net/browse/HFP-2830) and it's a pity that the pull request seems to have been ignored completely instead of taking in into consideration and working with it when creating the designs.

Now not only do we have this back and forth -- clearly, if pull-requests are handled that way it's more work and hassle than doing everything on your own -- but we're also antagonizing the people who funded this work back in the days. They needed that feature themselves, they asked whether it was fine to do it that way beforehand because they explicitly did not want to run a fork. And they will now be forced into one or they'll have to put in extra effort clean up the data structure of their contents. That's not a plus on business trustworthiness.

I guess I'll strip down my solution to an inferior one now ...

**Comment by otacke** [comment]
@simondate I have no good answer for your question about the different language, because I am not sure which you mean in particular. I just had a look at semantics.json and .en.json and noticed differences, so even those have not been kept in sync and they were inconsistent even before merging in my changes. Since people usually base their new translations on .en.json, that may already have caused some diverging. Merging in my pull request may have introduced some more differences, because it was lying around for over a year and I didn't keep up with all the new language files and changes. Not sure if you rebased the pull request branch first or just merged in the changes. In the latter case, that may have overwritten some changes that had been done meanwhile.

AAAND sorry for my previous comment. The message holds, but I should not shoot messengers. Sorry again. I have removed the extra behavioral settings. That was indeed a leftover. I'll now go for the other changes related to the proceed button in CP and IV, move that around and silently keep the feature available for all the other content types for the future instead of removing it completely ;-)

File: language/.en.json
Comment (simondate) [reply language/.en.json:61]: I think Forces is the correct word to use here (as a native speaker). I'll get Jelena's second opinion.

**Review by simondate — CHANGES_REQUESTED** [review]
I've found that if you use IV and complete the content type and submit the score but then press "replay the video". You are unable to resubmit your score and continue with the course.

I don't know if we should design it so that when you submit and restart you are able to resubmit or if you are just rewatching the video (in which case we should have the proceed button in the top right like we do with CP). @jelenamilinovic what do you think?

File: scripts/libraryScreen.js
Comment (otacke) [reply scripts/libraryScreen.js:206]: @simondate Well spotted. I missed that one when moving that stuff into a function.

File: scripts/libraryScreen.js
Comment (otacke) [reply scripts/libraryScreen.js:219]: @simondate Okay, definitely should not have worked that late ...

**Review by otacke — COMMENTED** [review]
> I've found that if you use IV and complete the content type and submit the score but then press "replay the video". You are unable to resubmit your score and continue with the course.

@simondate Oh, you're absolutely right. That's the usual IV/submit screen behavior, but now that the proceed button has been moved into the overlay instead of being in the upper right corner, you cannot proceed anymore once you're out of interactions to complete (which will reset the submit button when completed one). Didn't think of that.

> I don't know if we should design it so that when you submit and restart you are able to resubmit or if you are just rewatching the video (in which case we should have the proceed button in the top right like we do with CP). @jelenamilinovic what do you think?

@Jelena I am also not sure what behavior would be best.

File: scripts/libraryScreen.js
Comment (simondate) [reply scripts/libraryScreen.js:219]: ☕

**Review by simondate — APPROVED** [review]

**Pull request merged**
